### PR TITLE
[MINI-5811] delete file when failed download

### PR DESCRIPTION
**Pull request by shohiebsense**
# Description
Instead of checking connectivity first, that would cause create a hole where the device is connected to the internet first, then it goes offline on the file explorer -> save. 

Using this library as using the `contentResolver delete()` throws an [unsupported operation exception](https://stackoverflow.com/questions/63579633/deleting-a-file-given-a-storage-access-framework-uri-and-using-content-resolver)

Also if it is using traditional `File()` to delete requires Runtime Storage Permission.

## Links
MINI-5811

# Checklist
- [x] I have read the [contributing guidelines](https://github.com/rakutentech/android-miniapp/blob/master/.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](https://github.com/rakutentech/android-miniapp/blob/master/.github/CONTRIBUTING.md#sdk-development-learning-path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: miniapp/src/test/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileDownloaderSpec.kt [hunk 1]
```diff
@@ -45,9 +45,74 @@ class MiniAppFileDownloaderSpec {
             "ABzQ+pjAAAAC0lEQVQI12NgQAAAAAwAAeQ06mYAAAAASUVORK5CYII="
     private val testMimeType = "image/jpeg"
     private val testExpectedBytes = byteArrayOf(
-        -119, 80, 78, 71, 13, 10, 26, 10, 0, 0, 0, 13, 73, 72, 68, 82,
-        0, 0, 0, 3, 0, 0, 0, 3, 8, 0, 0, 0, 0, 115, 67, -22, 99, 0, 0, 0, 11, 73, 68, 65, 84, 8, -41, 99,
-        96, 64, 0, 0, 0, 12, 0, 1, -28, 52, -22, 102, 0, 0, 0, 0, 73, 69, 78, 68, -82, 66, 96, -126
```
Comment (khairul-alam-licon) [inline miniapp/src/test/java/com/rakuten/tech/mobile/miniapp/file/MiniAppFileDownloaderSpec.kt:50]: (optional) I think, we can still keep the old formation like this.

**Review by khairul-alam-licon — APPROVED** [review]
Looks good 👍 
Please add this fix in the change-log...

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 - **Feature:** Added `NativeEventType.MINIAPP_RECEIVE_JSON_INFO` to provide `Universal Bridge` for receiving messages from the HostApp to MiniApps.
 - **Feature:** Added `sendJsonToMiniApp` in `MiniAppView` and `MiniAppViewImpl` for receiving messages from the HostApp to MiniApps.
 - **Fix:** Prevent file creation with zero byte while downloading a file using `MiniAppFileDownloader` in Android 10.
+- **Fix:** Zero byte file deletion once there's an error during downloading a file
```
Comment (khairul-alam-licon) [inline CHANGELOG.md:9]:
```suggestion
- **Fix:** Prevent file creation with zero byte while downloading a file using `MiniAppFileDownloader` in offline mode.
```

Comment (shohiebsense) [reply CHANGELOG.md]: let's go for whatever :D

**Review by Climbatize — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,8 @@
 - **Feature:** Added `MiniAppMessageBridge.sendJsonToHostApp` interface to provide `Universal Bridge` for sending messages from a MiniApp to the HostApp.
 - **Feature:** Added `NativeEventType.MINIAPP_RECEIVE_JSON_INFO` to provide `Universal Bridge` for receiving messages from the HostApp to MiniApps.
 - **Feature:** Added `sendJsonToMiniApp` in `MiniAppView` and `MiniAppViewImpl` for receiving messages from the HostApp to MiniApps.
+- **Fix:** Prevent file creation with zero byte while downloading a file using `MiniAppFileDownloader` in Android 10.
+- **Fix:** Prevent file creation with zero byte while downloading a file using `MiniAppFileDownloader` in the offline mode.
```
Comment (rleojoseph) [inline CHANGELOG.md:9]:
```suggestion
- **Fix:** Prevent file creation with zero byte while downloading a file using `MiniAppFileDownloader` in Offline mode and also support for Android 10.
mode.
```

**Review by rleojoseph — APPROVED** [review]
Looks good to me :+1: Just a small suggestion in Changelog

**Comment by khairul-alam-licon** [comment]
Looks good 👍

**Review by munir-rakuten — APPROVED** [review]
Looks good.

**Pull request merged**